lib/model/model.go

```diff
@@ -1967,7 +1968,8 @@ func (s *indexSender) sendIndexTo(ctx context.Context) error {
 		if f.IsReceiveOnlyChanged() {
 			f.Version = protocol.Vector{}
 		}
-		f.LocalFlags = 0 // never sent externally
+		f.LocalFlags = 0     // never sent externally
+		f.BlockListKey = nil // never sent externally
```

Comment:
I wonder if we should just send this, and potentially reuse it for blockfinder/renames by having yet another indirection index?

Also, allows to do equivalence checks without having to load the blocklist, allowing to minimise db throughput in some cases.

Reply:
We could do that, I've many times thought about having a whole-file hash available for quicker comparisons. If so, it should probably be stricter specified than just "sha256 of whatever the protobuf marshaller happened to produce", something like sha256 of all the hashes and sizes sequentially or such. And we could produce it once and for all at scan time, so it wouldn't matter if it was slightly more expensive to produce...

Reply:
I made this a real field that is the sha of all the block hashes. Using it to optimize other things is a future thing.